src/create-route-map.js

```diff
@@ -43,6 +43,17 @@ function addRouteRecord (
   }
 
   if (route.children) {
+    // Warn if route is named and has a default child route.
+    // If users navigate to this route by name, the default child will
+    // not be rendered (GH Issue #629)
+    if (process.env.NODE_ENV !== 'production') {
+      if (route.name && route.children.some(child => /^\/?$/.test(child.path))) {
```

Comment:
We should also be checking for `route.children` here.

And in https://github.com/vuejs/vue-router/blob/next/examples/nested-routes/app.js#L29, it says
> an empty path will be treated as the default, e.g.

So maybe we can just use `child.path === ''` here?